appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy

```diff
@@ -248,6 +274,12 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 		assert dayActivityForGoal._links.self == null  // This is not a top level or embedded resource
 	}
 
+	private Goal findWeekGoal(expectedValues, activityCategoryUrl)
+	{
+		def expectedValuesEndOfWeek = (expectedValues.size() == 1) ? expectedValues[getCurrentShortDay()] : expectedValues["Sat"]
```

Comment:
A goal update always applies to the day or week at which it was updated. That implies that in case of multiple updates in that day or week, the last update applies. That's why this code finds the goal applicable to the last day of the week.